src/tests/ftest/util/soak_utils.py

```diff
@@ -673,67 +677,81 @@ def launch_reboot(self, pools, name, results, args):
+            status = True
+            self.dmg_command.system_query()
+            # wait for node to complete rebooting
+            if not wait_for_result(self.log, check_ping, 60, 5, True, host=reboot_host,
+                                   expected_ping=True, cmd_timeout=60, verbose=True):
+                self.log.error(f"<<<FAILED: {name} - {reboot_host} failed to reboot")
                 status = False
-        if status:
-            # reintegrate ranks
-            reintegrate_status = True
-            for pool in pools:
-                for rank in ranklist:
+            if not wait_for_result(self.log, check_ssh, 120, 2, True, hosts=reboot_host,
+                                   cmd_timeout=30, verbose=True):
+                self.log.error(f"<<<FAILED: {name} - {reboot_host} failed to reboot")
+                status = False
+            if status:
+                # issue a restart
+                self.log.info("<<<PASS %s: Issue systemctl restart daos_server on %s at %s>>>\n",
+                              self.loop, name, reboot_host, time.ctime())
+                cmd_results = run_remote(
+                    self.log, reboot_host, command_as_user("systemctl restart daos_server", "root"))
+                if cmd_results.passed:
+                    self.dmg_command.system_query()
+                    for pool in pools:
+                        pool.query()
+                    # wait server to be started
                     try:
-                        pool.reintegrate(rank)
-                        status = True
-                    except TestFail as error:
-                        self.log.error(
-                            f"<<<FAILED: dmg pool {pool.identifier} reintegrate failed on rank"
-                            "{rank}", exc_info=error)
+                        self.dmg_command.system_start(ranks=ranks)
+                    except CommandFailure as error:
+                        self.log.error("<<<FAILED:dmg system start failed", exc_info=error)
                         status = False
-                    reintegrate_status &= status
-                    if reintegrate_status:
-                        reintegrate_status &= wait_for_pool_rebuild(self, pool, name)
-                        status = reintegrate_status
-                    else:
-                        status = False
-    else:
-        self.log.error("<<<PASS %s: %s failed due to REBOOT failure >>>", self.loop, name)
-        status = False
+                    if status:
+                        # Check the servers are in joined state.
+                        all_joined = False
+                        retry = 0
+                        while not all_joined and retry < 10:
+                            all_joined = check_system_query_status(
+                                self.get_dmg_command().system_query())
+                            retry += 1
+                            time.sleep(10)
```

Comment:
Ideally this would be a 1 second wait.  If we required more time we could increase the retry count or use a time based while loop (`while time.time() < rejoin_timeout)`) that uses a timeout subject to the launch timeout multiplier.